contrib/generic_proxy/filters/network/source/match.h

```diff
@@ -24,10 +24,8 @@ using PropertyDataInputProto =
 class ServiceMatchDataInput : public Matcher::DataInput<Request> {
 public:
   Matcher::DataInputGetResult get(const Request& data) const override {
-    Matcher::DataInputGetResult result;
-    result.data_availability_ = Matcher::DataInputGetResult::DataAvailability::AllDataAvailable;
-    result.data_.emplace(data.host());
-    return result;
+    return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
+            std::string(data.host())};
```

Comment:
nit: maybe in the future we can used `UnionString` here to return string view or allocated string. But this could be a seperated pr.

Reply:
Yea, that is outside of the scope of this PR